docs/modules/developer-guides/pages/deprecating-steps.adoc

```diff
@@ -0,0 +1,127 @@
+----
+Composite: When I send HTTP $httpMethod to the relative URL '$relativeURL'
+!-- WARNING: The step "When I send HTTP $httpMethod to the relative URL '$relativeURL'" is deprecated and will be removed in VIVIDUS 0.6.0
+When I execute HTTP <httpMethod> request for resource with relative URL `<relativeURL>`
+----
+====
+
+== Find deprecated steps
+
+=== In log during/after test run
+
+[source]
+----
+2023-06-12 11:18:31,719 [batch-1-thread-3] INFO  org.vividus.log.LoggingStoryReporter - The step: "When I find greater than `1` JSON elements by `$.store.book` and for each element do
+|step                                                       |
+|Then number of JSON elements by JSON path `$.author` is = 1|" is deprecated and will be removed in VIVIDUS 0.6.0. Use step: "When I find greater than `1` JSON elements from `${json-context}` by `$.store.book` and for each element do
+|step                                                       |
+|Then number of JSON elements by JSON path `$.author` is = 1|"
+----
+
+=== In Allure report after test run
+
+* deprecated code step
+
+image::deprecated-code-step.png[Multiple external systems]
+
+* deprecated composite step
+
+image::deprecated-composite-step.png[Multiple external systems]
+
+=== Using print steps command (only for code steps)
+
+The xref:commons:cli.adoc#_print_available_steps[`printSteps`] command prints all available steps and mark deprecated ones:
+
+[source]
+----
+...
+vividus-plugin-rest-api              Then server `$hostname` supports secure protocols that $rule `$protocols`
+vividus-plugin-rest-api              Then size of decompressed response body is $comparisonRule `$sizeInBytes`
+vividus-plugin-rest-api  DEPRECATED  Then the connection is secured using $securityProtocol protocol
+vividus-plugin-rest-api  DEPRECATED  Then the number of the response headers with the name '$headerName' is $comparisonRule $value
+vividus-plugin-rest-api  DEPRECATED  Then the response body $comparisonRule '$content'
+...
+----
+
+=== In documentation (not for all steps)
+
+Some deprecated steps documented and can be found using search window.
+
+== Remove deprecated steps from tests
+
+Steps can only be removed from VIVIDUS in a release with breaking changes.
+Usually `MAJOR` version is incremented on breaking changes releases according to the https://semver.org/spec/v2.0.0.html[Semantic Versioning].
+But if project in initial development phase (has `0.y.z` version) incremented only `MINOR` version.
+
+image::semantic-versions.png[Semantic Versioning]
+
+Also, breaking changes releases usually have the `Removed` section with changes in the https://github.com/vividus-framework/vividus/releases[release history].
+
+image::removed-section.png[Removed section]
+
+In case of update to version with breaking changes necessary to check presence of all removed steps in test project and replace them manually if this cannot be done xref:commons:cli.adoc#_replace_deprecated_steps[automatically].
```

Comment:
```suggestion
In case of update to version with breaking changes it is necessary to check presence of all removed steps in test project and replace them manually if this cannot be done xref:commons:cli.adoc#_replace_deprecated_steps[automatically].
```